lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1598 @@
+        if eg_down_scaling_policies.__sizeof__() > 0:
+            eg_scaling.down = eg_down_scaling_policies
+
+    if eg_scaling.down is not None or eg_scaling.up is not None:
+        eg.scaling = eg_scaling
+
+
+def expand_scaling_policies(scaling_policies):
+    eg_scaling_policies = []
+
+    for policy in scaling_policies:
+        eg_policy = spotinst.aws_elastigroup.ScalingPolicy()
+
+        if policy.get('policy_name') is not None:
+            eg_policy.policy_name = policy.get('policy_name')
+
+        if policy.get('namespace') is not None:
+            eg_policy.namespace = policy.get('namespace')
+
+        if policy.get('metric_name') is not None:
+            eg_policy.metric_name = policy.get('metric_name')
+
+        if policy.get('dimensions') is not None:
+            eg_policy.dimensions = policy.get('dimensions')
+
+        if policy.get('statistic') is not None:
+            eg_policy.statistic = policy.get('statistic')
+
+        if policy.get('evaluation_periods') is not None:
+            eg_policy.evaluation_periods = policy.get('evaluation_periods')
+
+        if policy.get('period') is not None:
+            eg_policy.period = policy.get('period')
+
+        if policy.get('threshold') is not None:
+            eg_policy.threshold = policy.get('threshold')
+
+        if policy.get('cooldown') is not None:
+            eg_policy.cooldown = policy.get('cooldown')
+
+        eg_scaling_action = spotinst.aws_elastigroup.ScalingPolicyAction()
+
+        if policy.get('action_type') is not None:
+            eg_scaling_action.type = policy.get('action_type')
+
+        if policy.get('adjustment') is not None:
+            eg_scaling_action.adjustment = policy.get('adjustment')
+
+        if policy.get('min_target_capacity') is not None:
+            eg_scaling_action.min_target_capacity = policy.get('min_target_capacity')
+
+        if policy.get('max_target_capacity') is not None:
+            eg_scaling_action.max_target_capacity = policy.get('max_target_capacity')
+
+        if policy.get('target') is not None:
+            eg_scaling_action.target = policy.get('target')
+
+        if policy.get('minimum') is not None:
+            eg_scaling_action.minimum = policy.get('minimum')
+
+        if policy.get('maximum') is not None:
+            eg_scaling_action.maximum = policy.get('maximum')
+
+        if policy.get('unit') is not None:
+            eg_policy.unit = policy.get('unit')
+
+        if policy.get('operator') is not None:
+            eg_policy.operator = policy.get('operator')
+
+        eg_scaling_policies.append(eg_policy)
+
+    return eg_scaling_policies
+
+
+def main():
+    fields = dict(
+        account_id=dict(type='str'),
+        availability_vs_cost=dict(type='str'),
+        availability_zones=dict(type='list'),
+        block_device_mappings=dict(type='list'),
+        chef=dict(type='dict', required=False, default=None),
```

Comment:
`required=False` and `default=None` are defaults so you can eliminate them in every case below

Reply:
Fixed.